server/lib/chat/chat-models.ts

```diff
@@ -578,3 +579,105 @@ export async function findChannelByName(
     done()
   }
 }
+
+/**
+ * Returns a list of non-private chat channels that a user is not joined in, optionally filtered by
+ * a `searchStr`.
+ */
+export async function listChannels(
+  {
+    userId,
+    limit,
+    pageNumber,
+    searchStr,
+  }: {
+    userId: SbUserId
+    limit: number
+    pageNumber: number
+    searchStr?: string
+  },
+  withClient?: DbClient,
+): Promise<ChannelInfo[]> {
+  const { client, done } = await db(withClient)
+  try {
+    const query = sql`
+      WITH unjoined_channels AS (
+        SELECT DISTINCT(channel_id)
+        FROM channel_users
+        WHERE channel_id NOT IN (SELECT channel_id FROM channel_users WHERE user_id = ${userId})
+      )
+      SELECT c.*, COUNT(cu.user_id) AS user_count
```

Comment:
This count seems incredibly expensive and should probably be tracked in the channel itself or something. You should definitely look at the query plan for this because I cannot imagine it is good